Makefile

```diff
@@ -4,13 +4,19 @@ BIN_DIR = ./bin
 BUILD_DIR = ./build
 ARCH = $(shell uname -p)
 
+CAIRO_FILES = $(shell find src -type f -iname "*.cairo" -and -not -iname "test_*.cairo")
 CAIRO_PROGRAM = $(BUILD_DIR)/zerosync_compiled.json
 STARK_PARSER = $(BIN_DIR)/stark_parser
 RUST_TEST_LIB = $(BIN_DIR)/libzerosync_tests.dylib
 
-CAIRO_PROGRAM:
-	find src -type f \( -iname "*.cairo" -and -not -iname "test_*.cairo" \) \
-		-exec cairo-compile {} --cairo_path src > build/zerosync_compiled.json \;
+CAIRO_PROGRAMS:
+	@echo "Compiling Cairo files..."
+	@find src -type d | sed 's/src/build/g' | xargs mkdir -p
+	@for file in ${CAIRO_FILES}; do \
+		target=$$(echo "$${file%.cairo}_compiled.json" | sed 's/src/build/g'); \
```

Comment:
This creates the sub-directories in `build` folder to mimic `src` structure and add `_compiled` suffix to compiled program i.e. `build/stark_verifier/composer_compiled.json`